File: georgia_tech/sgl-eval-peterhj-genserver/backend/eval_next_backend/api_types/leaderboard.py
from typing import List, Dict
from pydantic import BaseModel

class LeaderboardTableRow(BaseModel):
    rank: int
    llm_name: str
    overall_score: float
    category_scores: Dict[str, float]

class LeaderboardResponse(BaseModel):
    category_names: List[str]
    table_rows: List[LeaderboardTableRow] 

File: georgia_tech/sgl-eval-peterhj-genserver/website/src/api_types/leaderboard.ts
export type LeaderboardTableRow = {
  rank: number;
  llm_name: string;
  overall_score: number;
  category_scores: Record<string, number>;
};

export type LeaderboardResponse = {
  category_names: string[];
  table_rows: LeaderboardTableRow[];
}; 